### PR TITLE
Add QED corrections into workflow

### PR DESCRIPTION
**Pull request by rymilton**
This PR adds the QED corrections to the data files formed in evaluate.py. This also adds the training scripts used to make the QED corrections model.

### CODE REVIEW

**Comment by rymilton** [comment]
Since the NoRad files don't have proper reco information (the data is all 0 except the weights), I modified `dataloader.py` to have a `use_reco` option that normalizes the weights using `num_pass_gen` instead of `num_pass_reco` and sets `pass_reco` to all False.

I added the flag `--QED_corrections` to train.py to load the MC files with and without radiation without the reco-level information.
Then while training, step 1 will be skipped since `pass_reco` is all False.

File: scripts/dataloader.py [hunk 1]
```diff
@@ -129,7 +136,13 @@ def prepare_dataset(self, file_names, pass_fiducial, pass_reco):
             ][start:end].astype(np.float32)
 
             self.weight.append(reco_e[:, -2].astype(np.float32))
-            self.pass_reco.append(reco_e[:, -1] == 1)
+            if not self.use_reco:
+                reco_e = np.nan_to_num(
+                    reco_e
+                )  # reco_e has a nan entry (reco_e[:,3]) when there's not proper reco info
```
Comment (rymilton) [inline scripts/dataloader.py]: Regarding this line, since `reco_e` is all 0 except the weights, [one variable will become np.nan](https://github.com/ViniciusMikuni/H1Unfold/blob/QEDcorrections/scripts/preprocess.py#L43)  which will throw an error in the `Dataset` class (see [this line](https://github.com/ViniciusMikuni/H1Unfold/blob/QEDcorrections/scripts/dataloader.py#L203)).

**Comment by rymilton** [comment]
I'm going to train a model for the QED corrections with this new code to make sure the results look reasonable before merging.

Once this is done, I'll move the model to the common directory and update the model path [here](https://github.com/ViniciusMikuni/H1Unfold/blob/QEDcorrections/scripts/utils.py#L907-L908), which is currently hardcoded to my directory.

File: scripts/dataloader.py (hunk 1 above)
Comment (rymilton) [reply scripts/dataloader.py]: Removed this behavior.

**Comment by rymilton** [comment]
During training, there's a variable called `num_steps_reco`. Its old value (`int(0.7* (self.mc.nmax + self.data.nmax))// hvd.size()// self.BATCH_SIZE`) did not work with the QED correction training since the model complained about running out of data during training. I'm assuming this 0.7 factor is the fraction of events that pass reconstruction, which is roughly 0.7. For QED corrections, we're using gen level with the pass_gen mask, so this should be changed to the fraction of events that pass generation cuts, which is 0.5. This is adjusted in [this line](https://github.com/ViniciusMikuni/H1Unfold/blob/QEDcorrections/scripts/omnifold.py#L350).
